### PR TITLE
Properly separate unit tests from system tests WRT subprocess

### LINKED ISSUE #601: Tests take too long
Just the tests in `test_runner.py` take 30s locally. All the tests combined take nearly 90 seconds

## Acceptance Criteria
`tox -e tests` takes less than 30s on most machines

### PR DESCRIPTION
**Pull request by callumforrester**
Fixes #601

### CODE REVIEW

**Comment by callumforrester** [comment]
Tagged @joeshannon for interest

**Comment by callumforrester** [comment]
See #601 for the old 20 slowest tests

```
================================ slowest 20 durations =================================
0.20s setup    tests/unit_tests/worker/test_task_worker.py::test_full_queue_raises_WorkerBusyError
0.13s setup    tests/unit_tests/worker/test_task_worker.py::test_plan_failure_recorded_in_active_task
0.12s setup    tests/unit_tests/worker/test_task_worker.py::test_clear_task_span_ok
0.12s setup    tests/unit_tests/worker/test_task_worker.py::test_begin_task_span_ok
0.11s setup    tests/unit_tests/worker/test_task_worker.py::test_submit_task_span_ok
0.11s call     tests/unit_tests/core/test_context.py::test_add_plan_from_module
0.11s call     tests/unit_tests/service/test_openapi.py::test_schema_updated
0.11s setup    tests/unit_tests/worker/test_task_worker.py::test_get_tasks_by_status[PENDING-expected_task_ids1]
0.11s setup    tests/unit_tests/worker/test_task_worker.py::test_does_not_allow_simultaneous_running_tasks
0.11s setup    tests/unit_tests/worker/test_task_worker.py::test_submit_before_start_pending
0.11s call     tests/unit_tests/test_cli.py::test_env_timeout
0.10s setup    tests/unit_tests/worker/test_task_worker.py::test_stop_span_ok
0.10s setup    tests/unit_tests/worker/test_task_worker.py::test_restart_leaves_task_pending
0.10s setup    tests/unit_tests/worker/test_task_worker.py::test_begin_task_blocks_until_current_task_set
0.10s setup    tests/unit_tests/worker/test_task_worker.py::test_clear_nonexistent_task
0.10s setup    tests/unit_tests/worker/test_task_worker.py::test_produces_worker_events[1]
0.10s setup    tests/unit_tests/worker/test_task_worker.py::test_clear_task
0.10s call     tests/unit_tests/service/test_interface.py::test_stomp_config
0.10s setup    tests/unit_tests/worker/test_task_worker.py::test_get_tasks_by_status[COMPLETE-expected_task_ids2]
0.10s setup    tests/unit_tests/worker/test_task_worker.py::test_worker_and_data_events_produce_in_order
```

File: src/blueapi/service/runner.py [hunk 1]
```diff
@@ -68,12 +70,9 @@ def reload(self):
 
     @start_as_current_span(TRACER)
     def start(self):
-        add_span_attributes(
-            {"_use_subprocess": self._use_subprocess, "_config": str(self._config)}
-        )
+        add_span_attributes({"_use_subprocess": True, "_config": str(self._config)})
```
[the remark this answers is not in the document]
Comment (DiamondJoseph) [reply src/blueapi/service/runner.py:73]: see below

**Review by DiamondJoseph — CHANGES_REQUESTED** [review]

**Comment by callumforrester** [comment]
Apologies, I mistakenly believed this was merged. I have rebased and updated the new tests to reflect this scheme. Unfortunately a new slow test has been introduced in the auth code since I did this PR. I suggest making that into a new issue since it only adds 2s to the test time and the acceptance criteria for #601 are still met. Current test times below:

```
======================================================================= slowest 20 durations ========================================================================
2.01s call     tests/unit_tests/service/test_authentication.py::test_poll_for_token_timeout
0.24s setup    tests/unit_tests/worker/test_task_worker.py::test_restart_leaves_task_pending
0.20s setup    tests/unit_tests/worker/test_task_worker.py::test_multi_stop
0.19s call     tests/unit_tests/service/test_openapi.py::test_schema_updated
0.18s setup    tests/unit_tests/worker/test_task_worker.py::test_stop_doesnt_hang
0.17s setup    tests/unit_tests/worker/test_task_worker.py::test_stop_is_idempontent_if_worker_not_started
0.16s setup    tests/unit_tests/worker/test_task_worker.py::test_submit_task
0.16s setup    tests/unit_tests/worker/test_task_worker.py::test_multi_start
0.16s teardown tests/unit_tests/service/test_interface.py::test_begin_task_no_task_id
0.15s setup    tests/unit_tests/worker/test_task_worker.py::test_restart
0.14s teardown tests/unit_tests/service/test_interface.py::test_get_device
0.14s call     tests/unit_tests/service/test_interface.py::test_begin_task
0.14s setup    tests/unit_tests/worker/test_task_worker.py::test_submit_multiple_tasks
0.13s call     tests/unit_tests/service/test_interface.py::test_get_tasks_by_status
0.13s setup    tests/unit_tests/worker/test_task_worker.py::test_worker_and_data_events_produce_in_order
0.12s setup    tests/unit_tests/worker/test_task_worker.py::test_stop_with_task_pending
0.12s call     tests/unit_tests/service/test_interface.py::test_stomp_config
0.12s teardown tests/unit_tests/service/test_interface.py::test_get_oidc_config
0.12s setup    tests/unit_tests/worker/test_task_worker.py::test_produces_worker_events[2]
0.12s setup    tests/unit_tests/worker/test_task_worker.py::test_full_queue_raises_WorkerBusyError
```

**Review by DiamondJoseph — APPROVED** [review]

**Pull request merged**